test/index.test.js

```diff
@@ -0,0 +1,312 @@
+      });
+    });
+  });
+
+  it('support multiple property level indexes', function(done) {
+    Product = db.define(modelName, {
+      prodName: {type: String, index: true},
+      prodPrice: {type: Number},
+      prodCode: {type: String, index: true},
+    });
+
+    db.automigrate(function(err) {
+      should.not.exist(err);
+
+      connector.getIndexes(connector.getDbName(connector), function(err, indexes) {
+        should.not.exist(err);
+        should.exist(indexes);
+
+        indexes = indexes.indexes;
+        var nameIndex = 'prodName_index';
+        var codeIndex = 'prodCode_index';
+
+        var nameIndexDoc = _.find(indexes, function(index) {
+          return index.name === nameIndex;
+        });
+
+        var codeIndexDoc = _.find(indexes, function(index) {
+          return index.name === codeIndex;
+        });
+
+        should.exist(codeIndexDoc);
+        should.exist(nameIndexDoc);
+        should.exist(codeIndexDoc.name);
+        should.exist(nameIndexDoc.name);
+        codeIndexDoc.name.should.equal(codeIndex);
+        nameIndexDoc.name.should.equal(nameIndex);
+        codeIndexDoc.def.fields[0]['prodCode'].should.equal('asc');
+        nameIndexDoc.def.fields[0]['prodName'].should.equal('asc');
+        done();
+      });
+    });
+  });
+
+  it('support composite indexes going same direction', function(done) {
+    Product = db.define(modelName, {
+      prodName: {type: String},
+      prodPrice: {type: Number},
+      prodCode: {type: String},
+    }, {
+      indexes: {
+        'price_code_index': {
+          keys: {
+            prodPrice: 1,
+            prodCode: 1,
+          },
+        },
+      },
+    });
+
+    db.automigrate('Product', function(err) {
+      connector.getIndexes(connector.getDbName(connector), function(err, indexes) {
+        indexes = indexes.indexes;
+        var indexName = 'price_code_index';
+
+        should.not.exist(err);
+        should.exist(indexes);
+
+        var index = _.find(indexes, function(index) {
+          return index.name === indexName;
+        });
+
+        should.exist(index);
+        should.exist(index.name);
+        index.name.should.equal(indexName);
+        index.def.fields[0]['prodPrice'].should.equal('asc');
+        index.def.fields[1]['prodCode'].should.equal('asc');
+        done();
+      });
+    });
+  });
+
+  it('throw error when composite indexes go opposite direction', function(done) {
+    Product = db.define(modelName, {
+      prodName: {type: String, index: true},
+      prodPrice: {type: Number, index: true},
+      prodCode: {type: String},
+    }, {
+      indexes: {
+        'code_price_index': {
+          keys: {
+            prodCode: 1,
+            prodPrice: -1,
+          },
+        },
+      },
+    });
+
+    db.automigrate('Product', function(err) {
+      should.exist(err);
+      should.equal(err.statusCode, 400);
+      should.equal(err.error, 'unsupported_mixed_sort');
+      done();
+    });
+  });
```

Comment:
Based on my previous comment, do we want to throw an error, or simply a warning? If they're half-way through an auto-update or auto-migration, having it abort mid-run could leave their database in a strange state, especially if it's for something that we can automatically handle with a warning message.